lhotse/recipes/earnings21.py

```diff
@@ -0,0 +1,177 @@
+
+    @misc{delrio2021earnings21,
+        title={Earnings-21: A Practical Benchmark for ASR in the Wild},
+        author={Miguel Del Rio and Natalie Delworth and Ryan Westerman and Michelle Huang and Nishchal Bhandari and Joseph Palakapilly and Quinten McNamara and Joshua Dong and Piotr Zelasko and Miguel Jette},
+        year={2021},
+        eprint={2104.11348},
+        archivePrefix={arXiv},
+        primaryClass={cs.CL}
+    }
+
+"""
+
+
+import logging
+import shutil
+import string
+import zipfile
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+from lhotse import validate_recordings_and_supervisions
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, urlretrieve_progress
+
+_DEFAULT_URL = (
+    "https://codeload.github.com/revdotcom/speech-datasets/zip/refs/heads/main"
+)
+
+
+def normalize(text: str) -> str:
+    # Remove all punctuation
+    text = text.translate(str.maketrans("", "", string.punctuation))
+    # Convert all upper case to lower case
+    text = text.lower()
+    return text
+
+
+def download_earnings21(
+    target_dir: Pathlike = ".",
+    force_download: Optional[bool] = False,
+    url: Optional[str] = _DEFAULT_URL,
+) -> Path:
+    """Download and untar the dataset.
+    :param target_dir: Pathlike, the path of the dir to store the dataset.
+        The extracted files are saved to target_dir/earnings21/
+        Please note that the github repository contains other additional datasets and
+        using this call, you will be downloading all of them and then throwing them out.
+    :param force_download: Bool, if True, download the tar file no matter
+        whether it exists or not.
+    :param url: str, the url to download the dataset.
+    :return: the path to downloaded and extracted directory with data.
+    """
+    logging.info(
+        "Downloading Earnings21 from github repository is not very efficient way"
+        + " how to obtain the corpus. You will be downloading other data as well."
+    )
+    target_dir = Path(target_dir)
+    target_dir.mkdir(parents=True, exist_ok=True)
+    extracted_dir = target_dir / "earnings21"
+
+    zip_path = target_dir / "speech-datasets-main.zip"
+
+    completed_detector = extracted_dir / ".lhotse-download.completed"
+    if completed_detector.is_file():
+        logging.info(f"Skipping - {completed_detector} exists.")
+        return extracted_dir
+
+    if force_download or not zip_path.is_file():
+        urlretrieve_progress(
+            url, filename=zip_path, desc="Getting speech-datasets-main.zip"
+        )
+
+    shutil.rmtree(extracted_dir, ignore_errors=True)
+
+    with zipfile.ZipFile(zip_path) as zip:
+        for f in zip.namelist():
+            if "earnings21" in f:
+                zip.extract(f, path=target_dir)
+
+    shutil.move(target_dir / "speech-datasets-main" / "earnings21", target_dir)
```

Comment:
Just found out that for python < 3.9, `shutil.move(src, dst)` has a [bug](https://bugs.python.org/issue39140) when `src` is of type `PosixPath`, causing the download to fail with the following error:

```python
13:55 $ lhotse download earnings21 /export/c07/draj/
2022-05-21 14:15:47,389 INFO [earnings21.py:64] Downloading Earnings21 from github repository is not very efficient way how to obtain the corpus. You will be downloading other data as well.
Traceback (most recent call last):
  File "/home/draj/anaconda3/envs/scale/bin/lhotse", line 33, in <module>
    sys.exit(load_entry_point('lhotse', 'console_scripts', 'lhotse')())
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/export/c07/draj/mini_scale_2022/lhotse/lhotse/bin/modes/recipes/earnings21.py", line 12, in earnings21
    download_earnings21(target_dir)
  File "/export/c07/draj/mini_scale_2022/lhotse/lhotse/recipes/earnings21.py", line 91, in download_earnings21
    shutil.move(
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/shutil.py", line 787, in move
    real_dst = os.path.join(dst, _basename(src))
  File "/home/draj/anaconda3/envs/scale/lib/python3.8/shutil.py", line 750, in _basename
    return os.path.basename(path.rstrip(sep))
AttributeError: 'PosixPath' object has no attribute 'rstrip'
```

This can be resolved by simply changing to `str(src)` in the function call. I will make this change when I create a PR next.